experimental-machinery/ablation/plume-util-ablation.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+#set -e
+set -u
+set -o pipefail
+
+# this script performs the ablation study in section 8.2 on the plume-util benchmark. Note that this script
+# doesn't terminate when errors are encountered, because grep will fail if there are no new warnings or
+# no old warnings that aren't issued, which would cause the script to terminate early
+
+# prereq: plume-util is checked out in the current directory, with the no-lo, no-ra, no-af branches available locally.
+# prereq: run-always-call-on-plume-util.sh is in the current directory
+# prereq: the checker and the Checker Framework are built in the appropriate places
+
+run_ablation () {
+
+    variant_name=$1
+
+    cd plume-util
+
+    git checkout "${variant_name}"
+
+    cd ..
+
+    sh run-always-call-on-plume-util.sh &> "${variant_name}-results" || true
+
+    echo "new errors produced by ${variant_name} configuration:"
+
+    cat "${variant_name}-results" | grep "error:" | sort | uniq | wc -l
+
+    echo "old errors no longer produced by ${variant_name} configuration:"
+
+    grep "unneeded.suppression" "${variant_name}-results" | wc -l
+
+    echo "the result for the paper is the difference between these two numbers"
+
+}
+
```

Comment:
I think we need this check here:

if [ "x${JAVA8_HOME}" = "x" ]; then
  echo "Please set JAVA8_HOME to run the checker on Plume Util. Plume Util requires Java 8."
  exit 1
fi

Reply:
plume-util doesn't require Java 8. We only need that in the other benchmarks because they don't compile under Java 11, but plume-util should be fine on any recent Java version (I think).

Unless you tried running the script and it failed, but I doubt that because when I ran the experiment myself earlier I used Java 11.

Reply:
So, do we need to check whether java version is 8 or 11?

Reply:
I don't think a check is strictly necessary. If the current JDK isn't supported by the CF, then the CF gradle plugin (which plume-util uses to run our checker) will print a helpful error message, so I don't think we need to worry about it.